src/transformState.js

```diff
@@ -1,11 +1,30 @@
-'use strict';
+"use strict";
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
```

Comment:
The destructuring of 'action' assumes that every action object has 'extraData' and 'keysToRemove' properties. This can potentially cause 'undefined' errors if these properties are absent. Consider destructuring these properties inside their respective case blocks.